src/basic/virt.c

```diff
@@ -181,8 +179,43 @@ static int detect_vm_dmi(void) {
                                 return dmi_vendor_table[j].id;
                         }
         }
-#endif
+        return VIRTUALIZATION_NONE;
+}
 
+static int detect_vm_dmi(void) {
+#if defined(__i386__) || defined(__x86_64__) || defined(__arm__) || defined(__aarch64__)
+
+        int r;
+        r = detect_vm_dmi_vendor();
+        if (r == VIRTUALIZATION_AMAZON) {
```

Comment:
hmm, why check vendor *and* this bit? To distuingish amazon physical machines and amazon VMs?

Reply:
Yes, we need both checks. The DMI vendor information is identical between VMs and physical systems.